tests/RobotFramework/tests/tedge/call_tedge_config_list.robot

```diff
@@ -31,263 +34,342 @@ tedge config list --all
     ${set}    Execute Command    tedge config list
     Should Contain    ${set}    device.key_path=/etc/tedge/device-certs1/tedge-private-key.pem
 
-    #Undo the change by using the 'unset' command, value returns to default one
+    # Undo the change by using the 'unset' command, value returns to default one
     Execute Command
     ...    sudo tedge config unset device.key_path
     ${unset}    Execute Command    tedge config list
     Should Contain    ${unset}    device.key_path=/etc/tedge/device-certs/tedge-private-key.pem
 
 set/unset device.cert_path
-    #Changing device.cert_path
+    # Changing device.cert_path
     Execute Command
     ...    sudo tedge config set device.cert_path /etc/tedge/device-certs1/tedge-certificate.pem
     ${set}    Execute Command    tedge config list
     Should Contain    ${set}    device.cert_path=/etc/tedge/device-certs1/tedge-certificate.pem
 
-    #Undo the change by using the 'unset' command, value returns to default one
+    # Undo the change by using the 'unset' command, value returns to default one
     Execute Command
     ...    sudo tedge config unset device.cert_path
     ${unset}    Execute Command    tedge config list
     Should Contain    ${unset}    device.cert_path=/etc/tedge/device-certs/tedge-certificate.pem
 
 set/unset c8y.root_cert_path
-    Execute Command    sudo tedge config set c8y.root_cert_path /etc/ssl/certs1    #Changing c8y.root_cert_path
+    Execute Command    sudo tedge config set c8y.root_cert_path /etc/ssl/certs1    # Changing c8y.root_cert_path
     ${set}    Execute Command    tedge config list
     Should Contain    ${set}    c8y.root_cert_path=/etc/ssl/certs1
 
-    #Undo the change by using the 'unset' command, value returns to default one
+    # Undo the change by using the 'unset' command, value returns to default one
     Execute Command
     ...    sudo tedge config unset c8y.root_cert_path
     ${unset}    Execute Command    tedge config list
     Should Contain    ${unset}    c8y.root_cert_path=/etc/ssl/certs
 
 set/unset c8y.smartrest.templates
-    Execute Command    sudo tedge config set c8y.smartrest.templates 1    #Changing c8y.smartrest.templates
+    Execute Command    sudo tedge config set c8y.smartrest.templates 1    # Changing c8y.smartrest.templates
     ${set}    Execute Command    tedge config list
     Should Contain    ${set}    c8y.smartrest.templates=["1"]
 
-    #Undo the change by using the 'unset' command, value returns to default one
+    # Undo the change by using the 'unset' command, value returns to default one
     Execute Command
     ...    sudo tedge config unset c8y.smartrest.templates
     ${unset}    Execute Command    tedge config list
     Should Contain    ${unset}    c8y.smartrest.templates=[]
 
+set/unset c8y.topics
+    Execute Command    sudo tedge config set c8y.topics topic1,topic2    # Changing c8y.topics
+    ${set}    Execute Command    tedge config list
```

Comment:
Is there a test to cover setting the topics via environment variables? E.g. how to set multiple topics?

You can test this by using `env` before calling the tedge config command, for example:

```sh
${settings}=    Execute Command    sudo env TEDGE_C8Y_TOPICS=tedge/measurements/+,tedge/events/+ tedge config get c8y.topics
Should Contain ${settings} c8y.topics=["tedge/measurements/+", "tedge/events/+"]
```

Reply:
I would expand `tests/RobotFramework/tests/tedge/tedge_config_get.robot` file as it tests environment value. (Note: this file doesn't have full coverage of tedge config keys, but not my scope.)

Reply:
Though the important thing is to have a test to convert it mainly because I believe this is the first usage of an array value so we should ensure that we have test coverage to prevent accidental regression. Also the nice docs inform the users that they can configure the settings via env variables so they that brings it into scope. The location of the test is slightly less interesting :)

Reply:
The first array value is "c8y.smartrest.templates", however no test for its environmental variable.
Anyway, behind the code, "<cloud>.topics" and "c8y.smartrest.templates" use the same struct. One can cover the other :)

Reply:
Added